### PR TITLE
Check file size locally.

### PR DESCRIPTION
**Pull request by rbeer**
Same as #4478 without language .json.

Oh yeah; and I took the liberty to put `module.hideAlerts` over the internal functions. Helps (plugin authors), if you find the accessible functions up top. :)

### CODE REVIEW

**Comment by julianlam** [comment]
Ah, you can add language.json back, but it'll have to be the files in `en_GB` :smile:

**Comment by rbeer** [comment]
I'm confused - as usual. :smile: 

I've signed up to transifex, trying to add the translations this way, but couldn't find "uploads", at all. Being new to transifex, chances are good that I'm just missing something (e.g how to add a file, if that's possible), but https://www.transifex.com/nodebb/nodebb/content/ does show all files in `public/language/en_GB/` except [uploads.json](https://github.com/NodeBB/NodeBB/blob/master/public/language/en_GB/uploads.json), the one I tried to add manually then.

Which makes me think that the translation part is a separate issue.

p.s.
I didn't mean "without `langauage.json`", but rather "without the language json file". :)

**Comment by julianlam** [comment]
You've introduced new language strings, yes? You have to define them in the en_GB folder so they'll be translatable in Transifex

**Comment by julianlam** [comment]
Actually there don't seem to be any new language strings here.

**Comment by rbeer** [comment]
No, I didn't introduce new strings.
There is a file https://github.com/NodeBB/NodeBB/blob/master/public/language/en_GB/uploads.json, but no corresponding entry over at transifex. Its strings however are in use in public/src/modules/uploader.js (file changed in this PR) several times ([L55](https://github.com/NodeBB/NodeBB/blob/master/public%2Fsrc%2Fmodules%2Fuploader.js#L55), [L61](https://github.com/NodeBB/NodeBB/blob/master/public%2Fsrc%2Fmodules%2Fuploader.js#L61), [L84](https://github.com/NodeBB/NodeBB/blob/master/public%2Fsrc%2Fmodules%2Fuploader.js#L84))

In the other PR I simply copied that `uploads.json` from `en_GB/` to `de/` and translated the content. That then got rejected by the bot.

Again, I'm pretty sure now that this is an issue on its own. Changes in this PR don't rely on it.

**Comment by julianlam** [comment]
Oh, I see, you're right -- looks like that file is missing. I'll push it up to Transifex :smile:

**Comment by julianlam** [comment]
LGTM, @barisusakli do you mind double-checking and merging :smile:

File: public/src/modules/uploader.js [hunk 1]
```diff
@@ -107,9 +116,11 @@ define('uploader', ['csrf', 'translator'], function(csrf, translator) {
 		return response;
 	}
 
-	module.hideAlerts = function(modal) {
-		$(modal).find('#alert-status, #alert-success, #alert-error, #upload-progress-box').addClass('hide');
-	};
+	function hasValidFileSize(fileElement, maxSize) {
+		if (window.FileReader && maxSize) {
+			return fileElement.files[0].size <= maxSize * 1000;
+		}
```
Comment (barisusakli) [inline public/src/modules/uploader.js]: Need to return true here if window.FileReader  doesnt exist

Comment (rbeer) [inline public/src/modules/uploader.js]: [L69](https://github.com/NodeBB/NodeBB/pull/4479/files#diff-8492f30a9acf5c0dd88bc3f8707e343fR69) compares the return value explicitly to `false`. My logic is that only a truely too big file returns false, thus triggering the showAlert('error',..). Cases of maxSize not given or FileReader not present return `undefined`.

Comment (barisusakli) [inline public/src/modules/uploader.js]: Imo it makes more sense for the function to return true/false then you can remove that explicit === false check as well.

Comment (rbeer) [inline public/src/modules/uploader.js]: Should be ok now.

I'm thinking about implementing the same behavior for the composer plugins. composer-default doesn't use the upload-modal. What's your preference there? Shall the current way (simple open-file dialog) remain or would you be ok with using the modal there, too?
I'd use the modal, for UX consistency.

**Pull request merged**